### ISSUE TITLE
Allow setting role attributes on core blocks

### ISSUE THREAD

**Issue by carolinan** [WordPress/gutenberg#46002]
## What problem does this address?

Adding a `role` attribute, (for example `role="button"`) to a core block causes a block validation error.
Allowing the role attribute would increase the flexibility for theme and plugin authors. It would compliment and not be as limited as the HTML element selection option that is in the Advanced panel for supported blocks.

To add a `role` attribute right now you would need to filter the block (or even create a custom block).

## What is your proposed solution?

Similar to the aria label added in https://github.com/WordPress/gutenberg/pull/41744, the `role` should be a block support but not have a UI, because it is an advanced feature.

**Comment by gziolo** [comment]
It sounds like a very good idea that is aligned with how other advanced HTML attributes are handled.

**Comment by carolinan** [comment]
To make this issue actionable, there needs to be a final decision about if the `role` should be a block support or block attribute.

**Comment by benoitchantre** [comment]
A few days ago, we received a web accessibility audit which identified the lack of explicit landmark roles as an issue, despite the use of semantic HTML (header, footer, main).

It would be great to have a native solution rather than to rely on a render_block filter.

**Comment by webfolkcreative** [comment]
Agreed. I am also getting flagged for explicit landmark roles. Would love to see this introduced.
